src/libexpr/primops/fromYAML.cc

```diff
@@ -0,0 +1,155 @@
+// RYML: workaround for bug in amalgamate script - nix requires at least C++17
+#include <charconv>
+// enabling this macro has a performance penalty
+#define RYML_WITH_TAB_TOKENS
+#define RYML_SINGLE_HDR_DEFINE_NOW
+#include "../../rapidyaml/ryml_all.hpp"
+
+#include "primops.hh"
+#include "eval-inline.hh"
+
+
+namespace nix {
+
+struct NixContext {
+    EvalState & state;
+    const PosIdx pos;
+    std::string_view yaml;
+};
+
+static void s_error [[ noreturn ]] (const char* msg, size_t len, ryml::Location loc, void *nixContext)
```

Comment:
That's the signature required by rapidyaml... `loc` is not used because `msg` contains the location of the error.

Reply:
Ok, can you then please drop the parameter's name? it would signal to the reader "this is the signature i need, but i don't intend to use `loc` so it is *not* a mistake that it is never accessed."